src/hocs/withContract.js

```diff
@@ -0,0 +1,41 @@
+import React, {Component} from "react";
+import PropTypes from "prop-types";
+
+const withContract = (contract, mapContractToProps) => WrappedComponent =>
```

Comment:
Alternate pattern that I think might be easier to reason about (and prevent you from needing to use `let`): 
1. Set the default of `mapContractToProps` to be a function that does the `else` case of your ternary below
2. Throw an error if `mapContractToProps` is anything but a function.

Reply:
Also, what's the expected type for `contract`?

Reply:
1 + 2, 👍

3) `contract` is just an object with about 15 properties on it, always the same - should I just confirm `typeof object` or is there a way I can/should be more strict about that?